targets/esp8266/jshardware.c

```diff
@@ -692,8 +666,19 @@ bool jshIsUSBSERIALConnected() {
  */
 void jshUSARTKick(
     IOEventFlags device //!< The device to be kicked.
-  ) {
-  esp8266_uartTransmitAll(device);
+) {
+  // transmit anything that is sitting in the uart output buffers
+  if (device == EV_SERIAL1 || device == EV_SERIAL2) {
+    // Get the next character to transmit.  We will have reached the end when
+    // the value of the character to transmit is -1.
+    int c = jshGetCharToTransmit(device);
```

Comment:
The TX is not interrupt driven at all. It's possible to make it interrupt driven.